From d434638a683c01bd6decf8f41863044055237a50 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Fri, 27 Mar 2020 11:59:53 +0100
Subject: [PATCH 09/16] namehint: remember the direction from the upper levels

The current code resets the direction info for each level.
Simply remove this code.

Fixes: https://github.com/alsa-project/alsa-lib/issues/39

Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 src/control/namehint.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/control/namehint.c b/src/control/namehint.c
index 169bd4215fc5..ecd470f33d07 100644
--- a/src/control/namehint.c
+++ b/src/control/namehint.c
@@ -270,8 +270,6 @@ static int try_config(snd_config_t *config,
 	if (snd_config_search(cfg1, "type", &cfg) >= 0 &&
 	    snd_config_get_string(cfg, &str) >= 0 &&
 	    strcmp(str, "hw") == 0) {
-		list->device_input = -1;
-		list->device_output = -1;
 		if (snd_config_search(cfg1, "device", &cfg) >= 0) {
 			if (snd_config_get_integer(cfg, &dev) < 0) {
 				SNDERR("(%s) device must be an integer", buf);
-- 
2.16.4

